lexer.go

```diff
@@ -892,11 +911,7 @@ func (r *reader) peek() rune {
 		return unicode.ReplacementChar
 	}
 	v, w := rune(r.s[r.p.Offset]), 1
-	switch {
-	case v == 0:
-		r.w = w
-		return v // illegal UTF-8 encoding
-	case v >= 0x80:
```

Comment:
I think here should also consider the charset, including the `UNDERSCORE_CHARSET`, maybe you can check how MySQL implements the `lexer`  https://github.com/mysql/mysql-server/blob/beb865a960b9a8a16cf999c323e46c5b0c67f21f/sql/sql_lex.cc#L1359